runtime-modules/content-directory/src/lib.rs

```diff
@@ -1964,6 +1954,50 @@ impl<T: Trait> Module<T> {
         Ok(())
     }
 
+    /// Ensure provided indices of `existing_properties`  are valid indices of `Class` properties
+    pub fn ensure_schema_properties_are_valid_indices(
+        existing_properties: &BTreeSet<PropertyId>,
+        class_properties: &[Property<T>],
+    ) -> dispatch::Result {
+        let has_unknown_properties = existing_properties
+            .iter()
+            .any(|&prop_id| prop_id >= class_properties.len() as PropertyId);
+        ensure!(
+            !has_unknown_properties,
+            ERROR_CLASS_SCHEMA_REFERS_UNKNOWN_PROP_INDEX
+        );
+        Ok(())
+    }
+
+    // Create new `Schema` with existing and new properies provided
+    pub fn create_class_schema(
+        existing_properties: BTreeSet<PropertyId>,
+        class_properties: &[Property<T>],
+        new_properties: &[Property<T>],
+    ) -> Schema {
+        // Create new Schema with existing properies provided
+        let mut schema = Schema::new(existing_properties);
```

Comment:
Likewise here, no need to do this with mutation.

Reply:
Done https://github.com/iorveth/substrate-runtime-joystream/commit/13bb9f293120bb2f8a2ac9f0583b98eae1241292